pkg/controller/nodes/task/k8s/plugin_manager.go

```diff
@@ -115,6 +117,13 @@ func (e *PluginManager) AddObjectMetadata(taskCtx pluginsCore.TaskExecutionMetad
 		f := append(o.GetFinalizers(), finalizer)
 		o.SetFinalizers(f)
 	}
+
+	pod, casted := o.(*v1.Pod)
+	if casted {
+		if errs := validation.IsDNS1123Label(pod.Name); len(errs) > 0 {
+			pod.Name = utils.ConvertToDNS1123CompatibleString(pod.Name)
+		}
+	}
```

Comment:
What if casted is false. Should this be raising an error or is it expected that its always a v1.Pod castable object

Reply:
This is intended for fixing the pod name. I don't know if other request objects want the same treatment e.g. https://github.com/flyteorg/flyteplugins/blob/d0a6ee24404c73b2db6fe35dd59fd6a9e285f755/go/tasks/plugins/k8s/sagemaker/builtin_training.go#L37 TrainingJob.